### PR TITLE
fix #590

### LINKED ISSUE #590: Message "Invalid buffer name" in lightbulb
### Description
Error message when switching sessions

```
Error executing vim.schedule lua callback: Vim:E158: Invalid buffer name:
 3
stack traceback:
        [C]: in function 'sign_unplace'
        ...pack/packer/start/lspsaga.nvim/lua/lspsaga/lightbulb.lua:70: i
n function '_update_sign'
        ...pack/packer/start/lspsaga.nvim/lua/lspsaga/lightbulb.lua:94: i
n function 'render_action_virtual_text'
        ...pack/packer/start/lspsaga.nvim/lua/lspsaga/lightbulb.lua:126:
in function 'callback'
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1951: in function 'handle
r'
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1383: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

This error doesn't break any functionality, only annoying me with message.

I can't help with reproducing this issue, but i can explain what happens.

I'm using plugins to load sessions and while i go from one session to another, lspsaga begins async task related with  lightbulb feature, than buffers list changes and lspsaga trying to operate a buffer that not exists. 

Since I have a stable reproduction of the bug on my machine, I have come up with a solution:

```diff
diff --git a/lua/lspsaga/lightbulb.lua b/lua/lspsaga/lightbulb.lua
index 6e3aa97..d2196f9 100644
--- a/lua/lspsaga/lightbulb.lua
+++ b/lua/lspsaga/lightbulb.lua
@@ -77,6 +77,9 @@ local function _update_sign(bufnr, line)
 end

 local function render_action_virtual_text(bufnr, line, has_actions)
+  if vim.fn.bufexists(bufnr) == 0 then
+    return
+  end
   if not has_actions then
     if config.code_action_lightbulb.virtual_text then
       _update_virtual_text(bufnr, nil)
```

Should I create a PR to fix this?

### Details

<details><summary>Reproduce</summary>

I can't create a env to reproduce, but you can load my config from script 
(IT WILL OVERWRITE your nvim config and requires yay installed)
`sh -c "$(wget -O- https://raw.githubusercontent.com/coffebar/dotfiles/master/fetch-nvim-conf.sh)"`

I see this bug when switching my particular projects.
</details>

<details><summary>Environment</summary>

- nvim --version output: 
```
NVIM v0.8.1
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by builduser

Features: +acl +iconv +tui
```
- Operating system: 
Linux
- lspsaga commit: 
 201dbbd13d6bafe1144475bbcae9efde224e07ec
</details>

### PR DESCRIPTION
**Pull request by coffebar**

### CODE REVIEW

**Comment by glepnir** [comment]
use `nvim_buf_is_valid`

**Comment by coffebar** [comment]
changed, tested

File: lua/lspsaga/lightbulb.lua [hunk 1]
```diff
@@ -77,7 +77,7 @@ local function _update_sign(bufnr, line)
 end
 
 local function render_action_virtual_text(bufnr, line, has_actions)
-  if vim.fn.bufexists(bufnr) == 0 then
+  if not vim.api.nvim_buf_is_valid(bufnr) then
```
Comment (glepnir) [inline lua/lspsaga/lightbulb.lua:80]: just use api not `vim.api` .

Comment (coffebar) [reply lua/lspsaga/lightbulb.lua]: reasonable

**Comment by coffebar** [comment]
I didn't mention: updated with commit

**Pull request closed without merging**